css/src/components/form/input.scss

```diff
@@ -72,4 +80,69 @@ $input-focus-box-shadow-size: 0 0 0 0.125rem !default;
 			box-shadow: $input-focus-box-shadow-size $input-success-border-color;
 		}
 	}
+
+	&.input-icon-right {
+		padding-inline-start: $control-padding-horizontal;
+		padding-inline-end: $input-icon-size;
+
+		~ .icon {
+			inset-inline-start: unset !important;
+			inset-inline-end: 0;
+		}
+
+		.input[type='date'] {
+			position: relative;
+
+			&::-webkit-calendar-picker-indicator {
+				position: absolute;
+				width: 1.5em;
+				background: none;
+				color: transparent;
+				inset-inline-end: 0;
+				z-index: $zindex-dropdown;
+			}
+		}
+	}
+
+	&.input-icon,
+	&.input-icon-right {
+		padding-inline-start: $input-icon-size;
+
+		&.input-icon-right {
+			padding-inline-start: unset;
+		}
+
+		+ .icon {
+			position: absolute;
+			inset-block-start: 0;
+			inset-inline-start: 0;
+			width: $input-icon-size;
```

Comment:
Should sizes be done in `em` to ensure scaling? i.e. should we remove the variable declarations on 23-25 in favor of em to ensure this always scales, regardless of an input's size?

Reply:
Removed the icon font size variables and updated the `.icon` container font size instead. The svg defaults to 1em already. I think it's better to not force the user to use another font size helper class in this scenario to avoid weird accidental pairings (like input-sm with font-size-xxl). Adding `input sm/lg` should take care of the rest.